File: Offer/2267/main.go
package main

func firstUniqChar(s string) byte {
	if len(s) == 0 {
		return ' '
	}
	length := len(s)
	hashmap := make(map[byte]int)
	var i int
	var value byte
	var ok bool

	for i = 0; i < length; i++ {
		value = s[i]
		if _, ok = hashmap[value]; ok {
			hashmap[value] = length
		} else {
			hashmap[value] = i
		}
	}

	var minK byte = ' '
	var minV int = length
	for k, v := range hashmap {
		if v < minV {
			minK = k
			minV = v
		}
	}
	return minK
}

File: Offer/2267/main.py
from typing import Dict


class Solution:
    def firstUniqChar(self, s: str) -> str:
        if len(s) == 0:
            return " "
        hashmap: Dict[str, int] = {}
        length = len(s)
        for i in range(length):
            if hashmap.get(s[i]) != None:
                hashmap[s[i]] = length
            else:
                hashmap[s[i]] = i
        
        first = sorted(hashmap.items(), key=lambda item:item[1])[0]
        return first[0] if first[1] != length else " "

